src/com/amazon/ion/benchmark/IonSchemaUtilities.java

```diff
@@ -166,6 +168,77 @@ public static String parseTextConstraints(IonStruct constraintStruct, String key
         return constraint;
     }
 
+    /**
+     * This method is used for checking whether the constraint 'valid_value' has the annotation 'range'.
+     * If 'valid_value' has the annotation 'range', this method will return the range value in IonList format, else it will return null.
+     * @param constraintStruct is the Ion struct which contain the current constraint field.
+     * @return the result of checking whether constraint 'valid_value' contains the annotation 'range'.
+     * @throws Exception if error occurs when building the IonReader.
+     */
+    public static IonList checkValidRange(IonStruct constraintStruct) throws Exception {
+        IonList result = null;
+        IonSystem SYSTEM = IonSystemBuilder.standard().build();
+        IonLoader LOADER = SYSTEM.newLoader();
+        if (constraintStruct != null) {
+            try (IonReader reader = IonReaderBuilder.standard().build(constraintStruct)) {
+                reader.next();
+                reader.stepIn();
+                while (reader.next() != null) {
+                    if (reader.getFieldName().equals(KEYWORD_VALID_VALUE) && reader.getTypeAnnotations().length != 0) {
+                        // If the constraint contains the annotation 'range' it will return the range value in IonList format for the further data construction process.
+                        IonDatagram datagram = LOADER.load(reader);
```

Comment:
Can we add a check for `range` here? This doesn't seem to check the annotation `range::`. There could be invalid annotations `range::`, `abc::` etc as Ion schema allows for open content.

Reply:
> Can we add a check for `range` here? This doesn't seem to check the annotation `range::`. There could be invalid annotations `range::`, `abc::` etc as Ion schema allows for open content.

Yes, I assumed that the `range::` is the only possibility. I'll add it in the following commit.